Client (2.x) unable to reconnect after killing one cluster member

Client is initialized with 2 cluster members. After killing the cluster where it is connected, the client is unable to reconnect.

```
Aug  9 11:39:45 localhost debug: [INFO] com.hazelcast.cluster.ClusterManager Slf4jFactory.java:81 - [192.168.11.159]:5742 [event-test] #012#012Members [2] {#012#011Member [192.168.11.159]:5742 this#012#011Member [192.168.11.159]:5741#012}#012 
Aug  9 11:39:45 localhost debug: [INFO] com.hazelcast.cluster.ClusterManager Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] #012#012Members [2] {#012#011Member [192.168.11.159]:5742#012#011Member [192.168.11.159]:5741 this#012}#012 
Aug  9 11:39:46 localhost debug: [INFO] com.hazelcast.impl.PartitionManager Slf4jFactory.java:81 - [192.168.11.159]:5742 [event-test] Re-partitioning cluster data... Immediate-Tasks: 271, Scheduled-Tasks: 0 
Aug  9 11:39:47 localhost debug: [INFO] com.hazelcast.jmx.ManagementService Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] Hazelcast JMX agent enabled 
Aug  9 11:39:47 localhost debug: [INFO] com.hazelcast.impl.LifecycleServiceImpl Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] Address[192.168.11.159]:5741 is STARTED 
Aug  9 11:40:04 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Connection to Connection [1] [192.168.11.159:5742 -> 192.168.11.159:5742] is lost 
Aug  9 11:40:04 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Connection to Connection [1] [192.168.11.159:5742 -> 192.168.11.159:5742] is lost 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - SSE - HazelcastClient is CLIENT_CONNECTION_LOST 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.Connection Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Connection [Address[192.168.11.159]:5702] lost. Reason: java.io.EOFException[null] 
Aug  9 11:40:04 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Connection to Connection [0] [192.168.11.159:5702 -> 192.168.11.159:5702] is lost 
Aug  9 11:40:04 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Connection to Connection [0] [192.168.11.159:5702 -> 192.168.11.159:5702] is lost 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - SSE - HazelcastClient is CLIENT_CONNECTION_LOST 
Aug  9 11:40:04 localhost debug: [WARN] com.hazelcast.nio.ReadHandler Slf4jFactory.java:83 - [192.168.11.159]:5701 [test] hz._hzInstance_1_test.IO.thread-1 Closing socket to endpoint Address[192.168.11.159]:5702, Cause:java.io.EOFException 
Aug  9 11:40:04 localhost error: [WARN] com.hazelcast.nio.ReadHandler Slf4jFactory.java:83 - [192.168.11.159]:5701 [test] hz._hzInstance_1_test.IO.thread-1 Closing socket to endpoint Address[192.168.11.159]:5702, Cause:java.io.EOFException 
Aug  9 11:40:04 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - API - Connection to Connection [2] [192.168.11.159:5742 -> 192.168.11.159:5742] is lost 
Aug  9 11:40:04 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - API - Connection to Connection [2] [192.168.11.159:5742 -> 192.168.11.159:5742] is lost 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - API - HazelcastClient is CLIENT_CONNECTION_LOST 
Aug  9 11:40:04 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - API - Connection to Connection [1] [192.168.11.159:5702 -> 192.168.11.159:5702] is lost 
Aug  9 11:40:04 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - API - Connection to Connection [1] [192.168.11.159:5702 -> 192.168.11.159:5702] is lost 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.SocketAcceptor Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] 5741 is accepting socket connection from /192.168.11.34:32929 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.SocketAcceptor Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] 5741 is accepting socket connection from /192.168.11.34:32930 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.ConnectionManager Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] 5741 accepted socket connection from /192.168.11.34:32929 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.Connection Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] Connection [Address[192.168.11.159]:5742] lost. Reason: java.io.EOFException[null] 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.ConnectionManager Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] 5741 accepted socket connection from /192.168.11.34:32930 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - API - HazelcastClient is CLIENT_CONNECTION_LOST 
Aug  9 11:40:04 localhost debug: [WARN] com.hazelcast.nio.ReadHandler Slf4jFactory.java:83 - [192.168.11.159]:5741 [event-test] hz._hzInstance_2_event-test.IO.thread-1 Closing socket to endpoint Address[192.168.11.159]:5742, Cause:java.io.EOFException 
Aug  9 11:40:04 localhost error: [WARN] com.hazelcast.nio.ReadHandler Slf4jFactory.java:83 - [192.168.11.159]:5741 [event-test] hz._hzInstance_2_event-test.IO.thread-1 Closing socket to endpoint Address[192.168.11.159]:5742, Cause:java.io.EOFException 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.SocketAcceptor Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] 5701 is accepting socket connection from /192.168.11.34:55080 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.ConnectionManager Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] 5701 accepted socket connection from /192.168.11.34:55080 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.SocketAcceptor Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] 5701 is accepting socket connection from /192.168.11.34:55081 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.ConnectionManager Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] 5701 accepted socket connection from /192.168.11.34:55081 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.impl.ClientHandlerService Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] received auth from Connection [/192.168.11.34:32930 -> null] live=true, client=true, type=CLIENT, successfully authenticated 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.impl.ClientHandlerService Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] received auth from Connection [/192.168.11.34:55080 -> null] live=true, client=true, type=CLIENT, successfully authenticated 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.impl.ClientHandlerService Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] received auth from Connection [/192.168.11.34:32929 -> null] live=true, client=true, type=CLIENT, successfully authenticated 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.impl.ClientHandlerService Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] received auth from Connection [/192.168.11.34:55081 -> null] live=true, client=true, type=CLIENT, successfully authenticated 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.ConnectionManager Slf4jFactory.java:47 - SSE - Client is connecting to Connection [3] [192.168.11.159:5741 -> 192.168.11.159:5741] 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.ConnectionManager Slf4jFactory.java:47 - API - Client is connecting to Connection [3] [192.168.11.159:5741 -> 192.168.11.159:5741] 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - API - HazelcastClient is CLIENT_CONNECTION_OPENING 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.ConnectionManager Slf4jFactory.java:47 - API - Client is connecting to Connection [2] [192.168.11.159:5701 -> 192.168.11.159:5701] 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.ConnectionManager Slf4jFactory.java:47 - SSE - Client is connecting to Connection [1] [192.168.11.159:5701 -> 192.168.11.159:5701] 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - SSE - HazelcastClient is CLIENT_CONNECTION_OPENING 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - API - HazelcastClient is CLIENT_CONNECTION_OPENING 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - SSE - HazelcastClient is CLIENT_CONNECTION_OPENING 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - API - HazelcastClient is CLIENT_CONNECTION_OPENED 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.SocketConnector Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Could not connect to: /192.168.11.159:5702. Reason: ConnectException[Connection refused] 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.SocketConnector Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Could not connect to: /192.168.11.159:5702. Reason: ConnectException[Connection refused] 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.SocketConnector Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] Could not connect to: /192.168.11.159:5742. Reason: ConnectException[Connection refused] 
Aug  9 11:40:04  debug: last message repeated 2 times
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.SocketConnector Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Could not connect to: /192.168.11.159:5702. Reason: ConnectException[Connection refused] 
Aug  9 11:40:04 localhost debug: [WARN] com.hazelcast.nio.ConnectionMonitor Slf4jFactory.java:83 - [192.168.11.159]:5741 [event-test] Removing connection to endpoint Address[192.168.11.159]:5742 Cause => java.net.ConnectException {Connection refused}, Error-Count: 5 
Aug  9 11:40:04 localhost error: [WARN] com.hazelcast.nio.ConnectionMonitor Slf4jFactory.java:83 - [192.168.11.159]:5741 [event-test] Removing connection to endpoint Address[192.168.11.159]:5742 Cause => java.net.ConnectException {Connection refused}, Error-Count: 5 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.cluster.ClusterManager Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] Removing Address Address[192.168.11.159]:5742 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.impl.PartitionManager Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] Starting to send partition replica diffs... 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.cluster.ClusterManager Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] #012#012Members [1] {#012#011Member [192.168.11.159]:5741 this#012}#012 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.SocketConnector Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Could not connect to: /192.168.11.159:5702. Reason: ConnectException[Connection refused] 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.nio.SocketConnector Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Could not connect to: /192.168.11.159:5702. Reason: ConnectException[Connection refused] 
Aug  9 11:40:04 localhost debug: [WARN] com.hazelcast.nio.ConnectionMonitor Slf4jFactory.java:83 - [192.168.11.159]:5701 [test] Removing connection to endpoint Address[192.168.11.159]:5702 Cause => java.net.ConnectException {Connection refused}, Error-Count: 5 
Aug  9 11:40:04 localhost error: [WARN] com.hazelcast.nio.ConnectionMonitor Slf4jFactory.java:83 - [192.168.11.159]:5701 [test] Removing connection to endpoint Address[192.168.11.159]:5702 Cause => java.net.ConnectException {Connection refused}, Error-Count: 5 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.cluster.ClusterManager Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Removing Address Address[192.168.11.159]:5702 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.impl.PartitionManager Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Starting to send partition replica diffs... 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.cluster.ClusterManager Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] #012#012Members [1] {#012#011Member [192.168.11.159]:5701 this#012}#012 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - API - HazelcastClient is CLIENT_CONNECTION_OPENED 
Aug  9 11:40:04 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - SSE - HazelcastClient is CLIENT_CONNECTION_OPENED 
Aug  9 11:40:07 localhost debug: [INFO] com.hazelcast.impl.PartitionManager Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] Total 0 partition replica diffs have been processed. 
Aug  9 11:40:07 localhost debug: [INFO] com.hazelcast.impl.PartitionManager Slf4jFactory.java:81 - [192.168.11.159]:5741 [event-test] Re-partitioning cluster data... Immediate-Tasks: 0, Scheduled-Tasks: 0 
Aug  9 11:40:07 localhost debug: [INFO] com.hazelcast.impl.PartitionManager Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Total 0 partition replica diffs have been processed. 
Aug  9 11:40:07 localhost debug: [INFO] com.hazelcast.impl.PartitionManager Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Re-partitioning cluster data... Immediate-Tasks: 0, Scheduled-Tasks: 0 
Aug  9 11:40:14 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's ping call within 10 seconds! 
Aug  9 11:40:14 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's ping call within 10 seconds! 
Aug  9 11:40:14 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [124] operation=GET_MEMBERS in 5 seconds. 
Aug  9 11:40:19 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [124] operation=GET_MEMBERS in 10 seconds. 
Aug  9 11:40:24 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's ping call within 10 seconds! 
Aug  9 11:40:24 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's ping call within 10 seconds! 
Aug  9 11:40:24 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [126] operation=GET_MEMBERS in 5 seconds. 
Aug  9 11:40:24 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [124] operation=GET_MEMBERS in 15 seconds. 
Aug  9 11:40:29 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [126] operation=GET_MEMBERS in 10 seconds. 
Aug  9 11:40:29 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [124] operation=GET_MEMBERS in 20 seconds. 
Aug  9 11:40:34 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's ping call within 10 seconds! 
Aug  9 11:40:34 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's ping call within 10 seconds! 
Aug  9 11:40:34 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [128] operation=GET_MEMBERS in 5 seconds. 
Aug  9 11:40:34 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [126] operation=GET_MEMBERS in 15 seconds. 
Aug  9 11:40:34 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [124] operation=GET_MEMBERS in 25 seconds. 
Aug  9 11:40:39 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [128] operation=GET_MEMBERS in 10 seconds. 
Aug  9 11:40:39 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [126] operation=GET_MEMBERS in 20 seconds. 
Aug  9 11:40:39 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [124] operation=GET_MEMBERS in 30 seconds. 
Aug  9 11:40:44 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's ping call within 10 seconds! 
Aug  9 11:40:44 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's ping call within 10 seconds! 
Aug  9 11:40:44 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's requests for 30 seconds. Assuming it is dead, closing the connection! 
Aug  9 11:40:44 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's requests for 30 seconds. Assuming it is dead, closing the connection! 
Aug  9 11:40:44 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Connection to Connection [1] [192.168.11.159:5701 -> 192.168.11.159:5701] is lost 
Aug  9 11:40:44 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Connection to Connection [1] [192.168.11.159:5701 -> 192.168.11.159:5701] is lost 
Aug  9 11:40:44 localhost debug: [INFO] com.hazelcast.nio.Connection Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Connection [Address[192.168.11.34]:55081] lost. Reason: java.io.EOFException[null] 
Aug  9 11:40:44 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - SSE - HazelcastClient is CLIENT_CONNECTION_LOST 
Aug  9 11:40:44 localhost debug: [INFO] com.hazelcast.nio.SocketAcceptor Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] 5701 is accepting socket connection from /192.168.11.34:55227 
Aug  9 11:40:44 localhost debug: [WARN] com.hazelcast.nio.ReadHandler Slf4jFactory.java:83 - [192.168.11.159]:5701 [test] hz._hzInstance_1_test.IO.thread-0 Closing socket to endpoint Address[192.168.11.34]:55081, Cause:java.io.EOFException 
Aug  9 11:40:44 localhost error: [WARN] com.hazelcast.nio.ReadHandler Slf4jFactory.java:83 - [192.168.11.159]:5701 [test] hz._hzInstance_1_test.IO.thread-0 Closing socket to endpoint Address[192.168.11.34]:55081, Cause:java.io.EOFException 
Aug  9 11:40:44 localhost debug: [INFO] com.hazelcast.nio.ConnectionManager Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] 5701 accepted socket connection from /192.168.11.34:55227 
Aug  9 11:40:44 localhost debug: [INFO] com.hazelcast.impl.ClientHandlerService Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Client {Connection [null -> Address[192.168.11.34]:55081] live=false, client=true, type=CLIENT} has been removed. 
Aug  9 11:40:44 localhost debug: [INFO] com.hazelcast.impl.ClientHandlerService Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] received auth from Connection [/192.168.11.34:55227 -> null] live=true, client=true, type=CLIENT, successfully authenticated 
Aug  9 11:40:44 localhost debug: [INFO] com.hazelcast.client.ConnectionManager Slf4jFactory.java:47 - SSE - Client is connecting to Connection [2] [192.168.11.159:5701 -> 192.168.11.159:5701] 
Aug  9 11:40:44 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - SSE - HazelcastClient is CLIENT_CONNECTION_OPENING 
Aug  9 11:40:44 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [128] operation=GET_MEMBERS in 15 seconds. 
Aug  9 11:40:44 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [131] operation=GET_MEMBERS in 5 seconds. 
Aug  9 11:40:44 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [126] operation=GET_MEMBERS in 25 seconds. 
Aug  9 11:40:44 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [124] operation=GET_MEMBERS in 35 seconds. 
Aug  9 11:40:45 localhost debug: [INFO] com.hazelcast.client.OutRunnable Slf4jFactory.java:47 - SSE - Thread[hz.client.1.OutThread,5,main]: Calling on disconnect Member [192.168.11.159]:5701 
Aug  9 11:40:47  debug: last message repeated 14 times
Aug  9 11:40:47 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's requests for 30 seconds. Assuming it is dead, closing the connection! 
Aug  9 11:40:47 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's requests for 30 seconds. Assuming it is dead, closing the connection! 
Aug  9 11:40:47 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Connection to Connection [2] [192.168.11.159:5701 -> 192.168.11.159:5701] is lost 
Aug  9 11:40:47 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Connection to Connection [2] [192.168.11.159:5701 -> 192.168.11.159:5701] is lost 
Aug  9 11:40:47 localhost debug: [INFO] com.hazelcast.nio.Connection Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Connection [Address[192.168.11.34]:55227] lost. Reason: java.io.EOFException[null] 
Aug  9 11:40:47 localhost debug: [WARN] com.hazelcast.nio.ReadHandler Slf4jFactory.java:83 - [192.168.11.159]:5701 [test] hz._hzInstance_1_test.IO.thread-1 Closing socket to endpoint Address[192.168.11.34]:55227, Cause:java.io.EOFException 
Aug  9 11:40:47 localhost error: [WARN] com.hazelcast.nio.ReadHandler Slf4jFactory.java:83 - [192.168.11.159]:5701 [test] hz._hzInstance_1_test.IO.thread-1 Closing socket to endpoint Address[192.168.11.34]:55227, Cause:java.io.EOFException 
Aug  9 11:40:47 localhost debug: [INFO] com.hazelcast.impl.ClientHandlerService Slf4jFactory.java:81 - [192.168.11.159]:5701 [test] Client {Connection [/192.168.11.34:55227 -> Address[192.168.11.34]:55227] live=false, client=true, type=CLIENT} has been removed. 
Aug  9 11:40:47 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - SSE - HazelcastClient is CLIENT_CONNECTION_LOST 
Aug  9 11:40:47 localhost debug: [INFO] com.hazelcast.client.ConnectionManager Slf4jFactory.java:47 - SSE - Unable to get alive cluster connection, try in 5,000 ms later, attempt 1 of 1. 
Aug  9 11:40:49 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [131] operation=GET_MEMBERS in 10 seconds. 
Aug  9 11:40:49 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [128] operation=GET_MEMBERS in 20 seconds. 
Aug  9 11:40:49 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [126] operation=GET_MEMBERS in 30 seconds. 
Aug  9 11:40:49 localhost debug: [INFO] com.hazelcast.client.ProxyHelper Slf4jFactory.java:47 - SSE - There is no response for Call [124] operation=GET_MEMBERS in 40 seconds. 
Aug  9 11:40:50 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's requests for 30 seconds. Assuming it is dead, closing the connection! 
Aug  9 11:40:50 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's requests for 30 seconds. Assuming it is dead, closing the connection! 
Aug  9 11:40:52 localhost debug: [WARN] com.hazelcast.client.OutRunnable Slf4jFactory.java:45 - SSE - Could not restore the connection 
Aug  9 11:40:52 localhost error: [WARN] com.hazelcast.client.OutRunnable Slf4jFactory.java:45 - SSE - Could not restore the connection 
Aug  9 11:40:52 localhost debug: [WARN] com.hazelcast.client.OutRunnable Slf4jFactory.java:45 - SSE - Reconnection: true. Interrupting and shutting down the client! 
Aug  9 11:40:52 localhost error: [WARN] com.hazelcast.client.OutRunnable Slf4jFactory.java:45 - SSE - Reconnection: true. Interrupting and shutting down the client! 
Aug  9 11:40:52 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - SSE - HazelcastClient is SHUTTING_DOWN 
Aug  9 11:40:52 localhost debug: [INFO] com.hazelcast.client.HazelcastClient Slf4jFactory.java:47 - SSE - HazelcastClient[1] is shutting down. 
Aug  9 11:40:52 localhost debug: [INFO] com.hazelcast.client.OutRunnable Slf4jFactory.java:47 - SSE - OutRunnable is finished ok. 
Aug  9 11:40:52 localhost debug: [INFO] com.hazelcast.client.OutRunnable Slf4jFactory.java:47 - SSE - OutRunnable is finished. 
Aug  9 11:40:53 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's requests for 30 seconds. Assuming it is dead, closing the connection! 
Aug  9 11:40:53 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's requests for 30 seconds. Assuming it is dead, closing the connection! 
Aug  9 11:40:56 localhost debug: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's requests for 30 seconds. Assuming it is dead, closing the connection! 
Aug  9 11:40:56 localhost error: [WARN] com.hazelcast.client.ConnectionManager Slf4jFactory.java:45 - SSE - Server didn't respond to client's requests for 30 seconds. Assuming it is dead, closing the connection! 
Aug  9 11:40:57 localhost debug: [INFO] com.hazelcast.client.ConnectionManager Slf4jFactory.java:47 - SSE - ConnectionManager shutdown 
Aug  9 11:40:57 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - SSE - HazelcastClient shutdown completed in 5110 ms. 
Aug  9 11:40:57 localhost debug: [INFO] com.hazelcast.client.LifecycleServiceClientImpl Slf4jFactory.java:47 - SSE - HazelcastClient is SHUTDOWN 
Aug  9 11:40:58 localhost debug: [INFO] com.hazelcast.client.InRunnable Slf4jFactory.java:47 - SSE - InRunnable is finished ok. 
Aug  9 11:40:58 localhost debug: [INFO] com.hazelcast.client.InRunnable Slf4jFactory.java:47 - SSE - InRunnable is finished.
```

--- a/hazelcast-client/src/main/java/com/hazelcast/client/OutRunnable.java
+++ b/hazelcast-client/src/main/java/com/hazelcast/client/OutRunnable.java
@@ -203,6 +203,13 @@ public class OutRunnable extends IORunnable {
         temp.remove(RECONNECT_CALL);
         reconnectionCalls.addAll(client.getListenerManager().getListenerCalls());
         queue.addAll(reconnectionCalls);
+        if (q.size() > 0) {
+	        Call call = q.poll();
+	        while (call != null) {
+	        	queue.offer(call);
+	        	call = q.poll();
+	        }
+        }
         temp.drainTo(queue);
         queue.addAll(callMap.values());
     }

